src/runtime/pytuple.cs

```diff
@@ -68,12 +71,22 @@ public PyTuple(PyObject[] items)
         {
             int count = items.Length;
             obj = Runtime.PyTuple_New(count);
-            for (var i = 0; i < count; i++)
+            if (obj == IntPtr.Zero)
+            {
+                throw new PythonException();
+            }
+            for (int i = 0; i < count; i++)
             {
                 IntPtr ptr = items[i].obj;
                 Runtime.XIncref(ptr);
-                Runtime.PyTuple_SetItem(obj, i, ptr);
-                Runtime.CheckExceptionOccurred();
+                if (Runtime.PyTuple_SetItem(obj, i, ptr) != 0)
+                {
+                    for (int j = i; j >= 0; j--)
+                    {
+                        Runtime.XDecref(items[j].obj);
+                    }
```

Comment:
We should rather dec-ref `obj` I think, which should drag the already set elements along with it, shouldn't we?